gdx/src/com/badlogic/gdx/graphics/g3d/loader/ObjLoader.java

```diff
@@ -439,4 +420,61 @@ public ModelMaterial getMaterial (final String name) {
 		materials.add(mat);
 		return mat;
 	}
-}
+
+	private static class ObjMaterial {
+		String materialName = "default";
+		Color ambientColor;
+		Color diffuseColor;
+		Color specularColor;
+		float opacity;
+		float shininess;
+		String alphaTexFilename;
+		String ambientTexFilename;
+		String diffuseTexFilename;
+		String shininessTexFilename;
+		String specularTexFilename;
+
+		public ObjMaterial() {
+			reset();
+		}
+
+		public ModelMaterial build () {
+			ModelMaterial mat = new ModelMaterial();
+			mat.id = materialName;
+			mat.ambient = ambientColor == null ? null : new Color(ambientColor);
+			mat.diffuse = new Color(diffuseColor);
+			mat.specular = new Color(specularColor);
+			mat.opacity = opacity;
+			mat.shininess = shininess;
+			addTexture(mat, alphaTexFilename, ModelTexture.USAGE_TRANSPARENCY);
+			addTexture(mat, ambientTexFilename, ModelTexture.USAGE_AMBIENT);
+			addTexture(mat, diffuseTexFilename, ModelTexture.USAGE_DIFFUSE);
+			addTexture(mat, specularTexFilename, ModelTexture.USAGE_SPECULAR);
+			addTexture(mat, shininessTexFilename, ModelTexture.USAGE_SHININESS);
+
+			return mat;
+		}
+
+		private void addTexture (ModelMaterial mat, String texFilename, int usage) {
+			if (texFilename != null) {
+				ModelTexture tex = new ModelTexture();
+				tex.usage = usage;
+				tex.fileName = texFilename;
+				mat.textures.add(tex);
```

Comment:
textures array is not initialized. It will throw null exception

Reply:
As far as I can tell, yes, this is a problem. Other parts of the g3d package check if the `textures` field is null before using it.

Reply:
@xpenatan Nice catch! I forgot this line:
`if (mat.textures == null) mat.textures = new Array<ModelTexture>(1);`